GeorgianMemenator/Scenes/CatTagsList/CatTagsListViewModel.swift

```diff
@@ -8,11 +8,40 @@
 
 import SwiftUI
 import Foundation
+import Combine
 
 class CatTagsListViewModel: ObservableObject {
-    @Published var tags: [CatTagDetail] = [
-        CatTagDetail(name: "Sphynx cat"),
-        CatTagDetail(name: "Munchkin cat"),
-        CatTagDetail(name: "Bengal cat")
-    ]
+    private let catTagsService: CatListServing
+    
+    @Published var tags: [CatTag] = []
+    
+    private var cancellables = Set<AnyCancellable>()
+    
+    init(catTagsService: CatListServing) {
+        self.catTagsService = catTagsService
+    }
+    
+    deinit {
+        cancellables.removeAll()
+    }
+    
+    func initialFetch() {
+        getCatTags()
+    }
+    
+    func getCatTags() {
+        
+        catTagsService
+            .getCatTagss()
+            .receive(on: DispatchQueue.main)
+            .sink(
+            receiveCompletion: { [weak self] resp in
```

Comment:
Once you remove all logic from `recieveCompletion` closure, `[weak self]` is not needed any more. Can be safely removed. @georgemakadze